packages/ncids-js/src/components/nci-header/utils/mobile-menu/mobile-menu.ts

```diff
@@ -0,0 +1,456 @@
+		);
+
+		this.langCode = document.documentElement.lang;
+		this.initialize();
+	}
+
+	/**
+	 * Unregistgers mobile menu and remove event listeners
+	 *
+	 * @public
+	 */
+	public unregister(): void {
+		this.mobileButton.removeEventListener(
+			'click',
+			this.menuEventListener,
+			true
+		);
+		this.mobileClose.removeEventListener('click', this.menuEventListener, true);
+		this.mobileOverlay.removeEventListener(
+			'click',
+			this.menuEventListener,
+			true
+		);
+		document.removeEventListener('keydown', this.escapeKeyPressListener, false);
+
+		this.mobileOverlay.remove();
+		this.mobileClose.remove();
+		this.mobileNav.remove();
+	}
+
+	/**
+	 * Create mobile menu DOM items and place into header
+	 * @private
+	 */
+	private initialize(): void {
+		this.mobileNav = <HTMLElement>(
+			this.createDom('nav', ['nci-header-mobilenav'], [])
+		);
+		this.mobileOverlay = <HTMLElement>(
+			this.createDom('div', ['nci-header-mobilenav__overlay'], [])
+		);
+		const lang = this.langCode;
+		const ariaLabel = locale['close'][lang];
+		this.mobileClose = <HTMLElement>this.createDom(
+			'button',
+			['nci-header-mobilenav__close-btn'],
+			[
+				{
+					'aria-controls': 'nci-header-mobilenav',
+				},
+				{
+					'aria-label': ariaLabel,
+				},
+			]
+		);
+
+		this.mobileClose.addEventListener('click', this.menuEventListener, true);
+		this.mobileOverlay.addEventListener('click', this.menuEventListener, true);
+
+		this.mobileButton.addEventListener('click', this.menuEventListener, true);
+
+		this.mobileNav.append(this.mobileClose);
+		this.element.append(this.mobileNav);
+		this.element.append(this.mobileOverlay);
+
+		document.addEventListener('keydown', this.escapeKeyPressListener, false);
+		this.createCustomEvents();
+	}
+
+	/**
+	 * Handle Menu Button Click
+	 * @private
+	 */
+	private async handleToggleMenu(event: Event): Promise<void> {
+		event.preventDefault();
+		this.toggleMenu();
+	}
+
+	/**
+	 * Handle Escape Key
+	 * @private
+	 */
+	private async handleEscapeKeypress(event: KeyboardEvent): Promise<void> {
+		if (this.activeMenu) {
+			const isEscapePressed = event.key === 'Escape';
+			if (isEscapePressed) {
+				this.toggleMenu();
+			}
+		}
+	}
+
+	/**
+	 * Toggle Menu and Overlay Items
+	 * @private
+	 */
+	private async toggleMenu(): Promise<void> {
+		const isActive = this.mobileNav.classList.contains('active');
+		if (isActive) {
+			this.activeMenu = false;
+			this.focusTrap.toggleTrap(false, this.mobileNav);
+			this.mobileNav.classList.remove('active');
+			this.mobileOverlay.classList.remove('active');
+
+			this.element.dispatchEvent(this.customEvents['close']);
+		} else {
+			this.activeMenu = true;
+			this.mobileNav.classList.add('active');
+			this.mobileOverlay.classList.toggle('active');
+			// pull data
+			const path = this.pageUrl;
+			this.lastClicked = path;
+			this.lastClicked = path;
+
+			const results = await this.adaptor.getInitialMenuId(path);
```

Comment:
This I think should be:
`const results = await this.adaptor.getNavigationLevel(this.adaptor.getInitialMenuId())`